### PR TITLE
CON-2086-new redundant page.

### PR DESCRIPTION
**Pull request by shomavg**

### CODE REVIEW

File: src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs [hunk 1]
```diff
@@ -203,18 +199,72 @@ public async Task<ActionResult> WhenToApplyChange(string hashedAccountId, string
...
+
+        [HttpGet]
+        [Route("{hashedApprenticeshipId}/details/statuschange/{changeType}/maderedundant", Name = "MadeRedundant")]
+        [OutputCache(CacheProfile = "NoCache")]
+        public async Task<ActionResult> HasApprenticeBeenMadeRedundant(string hashedAccountId, string hashedApprenticeshipId, ChangeStatusType changeType, DateTime? dateOfChange, WhenToMakeChangeOptions whenToMakeChange)
+        {
+            if (!await IsUserRoleAuthorized(hashedAccountId, Role.Owner, Role.Transactor))
+                return View("AccessDenied");
+
+            var response = await _orchestrator.MakeApprenticeRedundant(hashedAccountId, hashedApprenticeshipId, changeType, dateOfChange, whenToMakeChange, OwinWrapper.GetClaimValue(@"sub"), null);
```
Comment (hkf-tech) [inline src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs:213]: just a personal opinion, can we rename the method MakeApprenticeRedundant to something like GetRedundantViewModel. As MakeApprenticeRedundant feels like it is going to make the apprentice redundant.

File: src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs [hunk 2]
```diff
@@ -203,18 +199,72 @@ public async Task<ActionResult> WhenToApplyChange(string hashedAccountId, string
...
+                var viewResponse = await _orchestrator.MakeApprenticeRedundant(hashedAccountId, hashedApprenticeshipId, model.ChangeType.Value,
+                    model.DateOfChange.DateTime, model.WhenToMakeChange, OwinWrapper.GetClaimValue(@"sub"), model.MadeRedundant);
+                
+                return View(new OrchestratorResponse<RedundantApprenticeViewModel>
+                {
+                    Data = viewResponse.Data
+                });
+            }
+
+            //var response = await _orchestrator.ValidateWhenToApplyChange(hashedAccountId, hashedApprenticeshipId, model);
```
Comment (hkf-tech) [inline src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs:239]: remove the not required commented code.

File: src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs [hunk 3]
```diff
@@ -203,18 +199,72 @@ public async Task<ActionResult> WhenToApplyChange(string hashedAccountId, string
...
+        [Route("{hashedApprenticeshipId}/details/statuschange/{changeType}/maderedundant", Name = "MadeRedundant")]
+        [OutputCache(CacheProfile = "NoCache")]
+        public async Task<ActionResult> HasApprenticeBeenMadeRedundant(string hashedAccountId, string hashedApprenticeshipId, ChangeStatusType changeType, DateTime? dateOfChange, WhenToMakeChangeOptions whenToMakeChange)
+        {
+            if (!await IsUserRoleAuthorized(hashedAccountId, Role.Owner, Role.Transactor))
+                return View("AccessDenied");
+
+            var response = await _orchestrator.MakeApprenticeRedundant(hashedAccountId, hashedApprenticeshipId, changeType, dateOfChange, whenToMakeChange, OwinWrapper.GetClaimValue(@"sub"), null);
+
+            return View(new OrchestratorResponse<RedundantApprenticeViewModel>
```
Comment (hkf-tech) [inline src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs:215]: Can we just do View(response)?

Comment (shomavg) [reply src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs:215]: i was just following the pattern used in the file.

File: src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs (hunk 1 above)
Comment (shomavg) [reply src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs:213]: 👍

File: src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs (hunk 2 above)
Comment (shomavg) [reply src/SFA.DAS.EAS.Web/Controllers/EmployerManageApprenticesController.cs:239]: 👍

**Pull request merged**